### ISSUE TITLE
Issue with column mapping

### ISSUE THREAD

**Issue by shahules786** [explodinggradients/ragas#309]
**Describe the bug**
Column map requires to input all required column names instead of only mapped columns. For example if I have a dataset with columns ["answer", "contexts"] and I need to use a metric like context_recall, Ideally I should only be passing `{"ground_truths":"answer"}` but here I have to pass `{"contexts":"context_v1","ground_truths":"ground_truths"}`

**Code to Reproduce**
```
from datasets import load_dataset
from ragas.metrics import context_recall
from ragas import evaluate

dataset = load_dataset("explodinggradients/WikiEval")
dataset = dataset['train'].map(lambda x : {'ground_truths':[x['answer']]})
ragas_score = evaluate(dataset=dataset,metrics=[context_recall],column_map={"contexts":"context_v1"})
```

**Comment by tinomaxthayil** [comment]
@shahules786 I can work on this issue

**Comment by tinomaxthayil** [comment]
Raised a PR for this issue.

**Comment by tinomaxthayil** [comment]
This is fixed with  [##331](https://github.com/explodinggradients/ragas/pull/331)
